typeclasses/rooms.py

```diff
@@ -74,4 +92,65 @@ def mv_delay(self):
         """Returns the movement delay for this room."""
         return self._TERRAINS[self.terrain]['delay']
 
+# https://stackoverflow.com/a/3221487/5244995
+REVERSE_EXIT_OFFSETS = dict((v,k) for k,v in EXIT_OFFSETS.iteritems())
+# TODO: diagonals (box-drawing only supports ╱, ╲, and ╳)
+CENTERS = {
+        'n': u'╷',
+        'e': u'╶',
+        's': u'╵',
+        'w': u'╴',
+      'e n': u'└',
+      'n s': u'│',
+      'n w': u'┘',
+      'e s': u'┌',
+      'e w': u'─',
+      's w': u'┐',
+    'e n w': u'┴',
+    'e s w': u'┬',
+    'n s w': u'┤',
+    'e n s': u'├',
+  'e n s w': u'┼'
+}
+SIDES = u'─'
+class Road(Room):
+    def at_object_creation(self):
+        super(Room, self).at_object_creation()
+        self.db.terrain = 'ROAD'
+
+    def map_tile(self, _map, centers=CENTERS, sides=SIDES):
+        directions = set()
+        for room, delta in get_directed_exits(self).items():
+            connects_to_roads = room.db.connects_to_roads or Room._TERRAINS[room.terrain].get('connects_to_roads', False)
+            if room.is_typeclass(Road) or connects_to_roads:
+                direction = REVERSE_EXIT_OFFSETS.get(delta)
+                if len(direction) == 1:
+                    directions.add(direction)
+        key = u' '.join(sorted(directions))
+        left = sides if 'w' in directions else u' '
+        right = sides if 'e' in directions else u' '
+        return left + centers.get(key, '?') + right
 
+BRIDGE_CENTERS = {
+    # there’s no proper box drawing characters
+    # for the plain directions
+        'n': u'║',
+        'e': u'═',
+        's': u'║',
+        'w': u'═',
+      'e n': u'╚',
+      'n s': u'║',
+      'n w': u'╝',
+      'e s': u'╔',
+      'e w': u'═',
+      's w': u'╗',
+    'e n w': u'╩',
+    'e s w': u'╦',
+    'n s w': u'╣',
+    'e n s': u'╠',
+  'e n s w': u'╬'
+}
+BRIDGE_SIDES = u'═'
```

Comment:
If there's a way to check if Unicode characters are able to be used, it would be good to have an alternative. If the player doesn't support Unicode characters, won't these either be stripped or display as boxes? It seems like Windows telnet doesn't support Unicode, same for some Android clients. Perhaps it doesn't matter?